app/client/src/sagas/ActionExecutionSagas.ts

```diff
@@ -658,72 +658,78 @@ function* confirmRunActionSaga() {
-  currentApp = currentApp || {};
-  yield put(executeApiActionRequest({ id: pageAction.id }));
-  const params: Property[] = yield call(
-    getActionParams,
-    pageAction.jsonPathKeys,
-  );
-  const appMode = yield select(getAppMode);
-  const viewMode = appMode === APP_MODE.PUBLISHED;
-  const executeActionRequest: ExecuteActionRequest = {
-    actionId: pageAction.id,
-    params,
-    viewMode,
-  };
-  AnalyticsUtil.logEvent("EXECUTE_ACTION", {
-    type: pageAction.pluginType,
-    name: pageAction.name,
-    pageId: pageId,
-    appId: currentApp.id,
-    onPageLoad: true,
-    appName: currentApp.name,
-    isExampleApp: currentApp.appIsExample,
-  });
-  const response: ActionApiResponse = yield ActionAPI.executeAction(
-    executeActionRequest,
-    pageAction.timeoutInMillisecond,
-  );
-  if (isErrorResponse(response)) {
-    yield put(
-      executeActionError({
-        actionId: pageAction.id,
-        error: response.responseMeta.error,
-        isPageLoad: true,
-      }),
-    );
-    PerformanceTracker.stopAsyncTracking(
+  try {
+    PerformanceTracker.startAsyncTracking(
       PerformanceTransactionName.EXECUTE_ACTION,
       {
-        failed: true,
+        actionId: pageAction.id,
       },
       pageAction.id,
+      PerformanceTransactionName.EXECUTE_PAGE_LOAD_ACTIONS,
     );
-  } else {
-    const payload = createActionExecutionResponse(response);
-    PerformanceTracker.stopAsyncTracking(
-      PerformanceTransactionName.EXECUTE_ACTION,
-      undefined,
-      pageAction.id,
+    const pageId = yield select(getCurrentPageId);
+    let currentApp: ApplicationPayload = yield select(getCurrentApplication);
+    currentApp = currentApp || {};
+    yield put(executeApiActionRequest({ id: pageAction.id }));
+    const params: Property[] = yield call(
+      getActionParams,
+      pageAction.jsonPathKeys,
     );
-    yield put(
-      executeApiActionSuccess({
-        id: pageAction.id,
-        response: payload,
-        isPageLoad: true,
-      }),
+    const appMode = yield select(getAppMode);
+    const viewMode = appMode === APP_MODE.PUBLISHED;
+    const executeActionRequest: ExecuteActionRequest = {
+      actionId: pageAction.id,
+      params,
+      viewMode,
+    };
+    AnalyticsUtil.logEvent("EXECUTE_ACTION", {
+      type: pageAction.pluginType,
+      name: pageAction.name,
+      pageId: pageId,
+      appId: currentApp.id,
+      onPageLoad: true,
+      appName: currentApp.name,
+      isExampleApp: currentApp.appIsExample,
+    });
+    const response: ActionApiResponse = yield ActionAPI.executeAction(
+      executeActionRequest,
+      pageAction.timeoutInMillisecond,
     );
-    yield take(ReduxActionTypes.SET_EVALUATED_TREE);
+    if (isErrorResponse(response)) {
+      yield put(
+        executeActionError({
+          actionId: pageAction.id,
+          error: _.get(response, "responseMeta.error", {
+            message: `The action with name "${pageAction.name}" has failed.`,
```

Comment:
```suggestion
            message: `The action "${pageAction.name}" has failed.`,
```

Also, by any chance do you get the error message from the server? If yes, is it possible to add that to the display here? Will make it easier for users to debug this.

Reply:
@mohanarpit We get this type of response from the server whenever the action is failing:

<img width="877" alt="Screenshot 2020-12-28 at 12 12 03 PM" src="https://user-images.githubusercontent.com/6636360/103194533-fd674600-4905-11eb-8667-60886f784667.png">

Reply:
Showing message like this now:

<img width="306" alt="Screenshot 2020-12-28 at 12 20 52 PM" src="https://user-images.githubusercontent.com/6636360/103194940-2e944600-4907-11eb-9ed8-e89cdf768e89.png">